soc/realtek/ameba/amebadplus/CMakeLists.txt

```diff
@@ -0,0 +1,37 @@
+# Copyright (c) 2024 Realtek Semiconductor Corp.
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_include_directories(${ZEPHYR_BASE}/drivers)
+zephyr_include_directories(.)
+
+zephyr_sources(soc.c)
+
+zephyr_linker_sources(SECTIONS boot_section.ld)
+set(SOC_LINKER_SCRIPT ${ZEPHYR_BASE}/include/zephyr/arch/arm/cortex_m/scripts/linker.ld CACHE INTERNAL "")
+
+set(PYTHON_SCRIPT_ARGS
```

Comment:
in cmake UPPER CASE means GLOBAL, and lower case means local, fix, and move this inside of the if()